File: day01/solution.py
def next_elf(file) -> list[int]:
  values = []
  for content in file:
    content = content.strip()
    if content == '':
      yield values
      values = []
    else:
      values.append(int(content.strip()))
  yield values


with open("sample", mode="rt") as file:
  calories = [ sum(values) for values in next_elf(file) if len(values) > 0]

calories.sort(reverse=True)
print(f"Max calories: {calories[0]}")
print(f"Top 3 sum: {sum(calories[0:3])}")

File: day01/solution.jl
function next(file)::Vector{UInt32}
  content::Vector{UInt32}=[]
  while (line = readline(file)) != "" && !eof(file)
    push!(content, parse(UInt32, line))
  end
  return content
end

calories::Vector{UInt32}=[]
open("inputs", "r") do file
  while !eof(file)
    entries = next(file)
    if !isempty(entries)
      push!(calories, reduce(+, entries))
    end
  end
end;

calories = sort(calories, rev=true)
println(string("Highest: ", calories[1]))

top3sum=reduce(+, calories[1:3])
println(string("Sum of top 3: ", top3sum))